app/src/org/commcare/android/database/user/UserDatabaseUpgrader.java

```diff
@@ -242,4 +311,45 @@ private void markSenseIncompleteUnsent(final SQLiteDatabase db) {
             storage.write(t);
         }
     }
+
+    private static boolean multipleInstalledAppRecords() {
+        int globalDbVersionNumber = CommCareApplication._().getGlobalDbVersion();
+        if (globalDbVersionNumber >= 3) {
+            // If the 1st multiple apps migration has already occurred, then app records will be
+            // stored in the db as an ApplicationRecord
+            SqlStorage<ApplicationRecord> storage =
+                    CommCareApplication._().getGlobalStorage(ApplicationRecord.class);
+            int count = 0;
+            for (ApplicationRecord r : storage) {
+                if (r.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
+                    count++;
+                }
+            }
+            return (count > 1);
+        } else {
+            // If the 1st multiple apps migration has NOT already occurred, then app records will
+            // be stored in the db as an ApplicationRecordV1
+            SqlStorage<ApplicationRecordV1> storage =
+                    CommCareApplication._().getGlobalStorage(ApplicationRecordV1.class);
+            int count = 0;
+            for (ApplicationRecordV1 r : storage) {
+                if (r.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
+                    count++;
+                }
+            }
+            return (count > 1);
+        }
+    }
+
+    public static ApplicationRecord getInstalledAppRecord() {
+        SqlStorage<ApplicationRecord> storage =
```

Comment:
@ctsims -- Want to confirm my intuition that any user db migrations will always occur after the global db migration, and that I can therefore assume that the version of app records in global storage at this point will always be ApplicationRecord, as opposed to ApplicationRecordV1